eth2util/rlp/rlp.go

```diff
@@ -101,8 +101,8 @@ func decodeLength(item []byte) (offset int, length int, err error) {
 	}
 
 	if prefix < 0xc0 {
-		length = int(prefix - 0xb7)
-		if length > 63 || length < 0 {
+		length = int(prefix - 0xb7) // length of the string in bytes in binary form
+		if length > 8 || length < 0 {
```

Comment:
<img width="810" alt="Screenshot 2023-04-11 at 15 50 41" src="https://user-images.githubusercontent.com/43061995/231132047-56cd6fe0-9861-4058-80a4-1d2fdca5c73f.png">
https://ethereum.org/en/developers/docs/data-structures-and-encoding/rlp/

Reply:
length == 8 includes negative integers when casting from uint, this should not be possible.

Reply:
On second thought, i think `length == 0` is also NOT possible since the range of the first byte is [`0xb8`, 0xbf]. So, the algorithm expects a positive length (>= 1).

Also, from this comment @corverroos , I believe the correct check should be:
```
if length >= 8 || length <= 0 { ...
```

Reply:
sure can do `if length > 8 || length <= 0`